app/experimenter/experiments/serializers/design.py

```diff
@@ -237,52 +267,80 @@ def update(self, instance, validated_data):
+):
+
+    class Meta:
+        model = RolloutPreference
+        list_serializer_class = PreferenceListSerializer
+        fields = ["id", "pref_name", "pref_type", "pref_value"]
+
+
+class ExperimentDesignPrefRolloutSerializer(
     PrefValidationMixin, ExperimentDesignBaseSerializer
 ):
     rollout_type = serializers.ChoiceField(choices=Experiment.ROLLOUT_TYPE_CHOICES)
-    addon_release_url = serializers.URLField(
-        max_length=400, allow_null=True, required=False
-    )
-
-    ROLLOUT_TYPE_FIELDS = {
-        Experiment.TYPE_PREF: ("pref_name", "pref_type", "pref_value"),
-        Experiment.TYPE_ADDON: ("addon_release_url",),
-    }
+    preferences = ExperimentDesignRolloutPreferenceSerializer(many=True)
 
     class Meta:
         model = Experiment
-        fields = (
-            "rollout_type",
-            "design",
-            "addon_release_url",
-            "pref_name",
-            "pref_type",
-            "pref_value",
-        )
+        fields = ("rollout_type", "design", "preferences")
 
     def validate(self, data):
         data = super().validate(data)
-        rollout_type = data["rollout_type"]
 
-        errors = {}
+        preferences = data["preferences"]
+
+        if data["rollout_type"] == Experiment.TYPE_PREF:
+            invalid_preferences = []
+            for preference in preferences:
+                pref_invalid = self.validate_multi_preference(preference)
 
-        for type_field in self.ROLLOUT_TYPE_FIELDS[rollout_type]:
-            if type_field not in data or not data[type_field]:
-                errors[type_field] = ["This field is required."]
+                invalid_preferences.append(pref_invalid)
 
-        if errors:
-            raise serializers.ValidationError(errors)
+            if any(invalid_preferences):
+                raise serializers.ValidationError({"preferences": invalid_preferences})
 
-        if data["rollout_type"] == Experiment.TYPE_PREF:
-            pref_invalid = self.validate_pref(
-                data["pref_type"], data["pref_value"], "pref_value"
+        if not self.is_pref_valid(preferences):
+            error_list = [{"pref_name": "Pref name needs to be unique"}] * len(
+                preferences
             )
-            if pref_invalid:
-                raise serializers.ValidationError(pref_invalid)
+            raise serializers.ValidationError({"preferences": error_list})
 
         return data
 
+    def update(self, instance, validated_data):
+        preferences_data = validated_data.pop("preferences", [])
+        instance = super().update(instance, validated_data)
+
+        if preferences_data:
+            existing_preference_ids = set(
+                instance.preferences.all().values_list("id", flat=True)
+            )
+            for preference_data in preferences_data:
+                preference_data["experiment"] = instance
+                RolloutPreference(**preference_data).save()
```

Comment:
**Opinion:**

This works, you can also do `RolloutPreference.objects.create(**preference_data)`.  I'm not sure it's all that different than this, I guess this just creates an instance and then immediately refcounts it to 0.